src/Neo/NeoSystem.cs

```diff
@@ -275,12 +275,24 @@ public void SuspendNodeStartup()
         /// <summary>
         /// Gets a snapshot of the blockchain storage.
         /// </summary>
-        /// <returns></returns>
+        /// <returns>An instance of <see cref="SnapshotCache"/></returns>
+        [Obsolete("This method is obsolete, use GetSnapshotCache instead.")]
         public SnapshotCache GetSnapshot()
         {
             return new SnapshotCache(store.GetSnapshot());
         }
 
+        /// <summary>
+        /// Gets a snapshot of the blockchain storage with an execution cache.
+        /// With the snapshot, we have the latest state of the blockchain, with the cache,
+        /// we can run transactions in a sandboxed environment.
+        /// </summary>
+        /// <returns>An instance of <see cref="SnapshotCache"/></returns>
+        public SnapshotCache GetSnapshotCache()
```

Comment:
This should be called `CreateSnapshotCache()`. Make more sense.

Reply:
may or may not, it was get snapshot, so its get snapshot cache. I dont have intention to change it that much.